src/main/java/ru/skypro/homework/controller/CommentController.java

```diff
@@ -104,7 +104,7 @@ public ResponseEntity<Void> deleteComments(@PathVariable("adId") Integer adId,
                     @ApiResponse(responseCode = "401", description = "Unauthorized")
             }
     )
-    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
+    @PreAuthorize("@commentServiceImpl.getEntity(#adId).author.email.equals(#auth.name) or hasRole('ADMIN')")
```

Comment:
```suggestion
    @PreAuthorize("@commentServiceImpl.getEntity(#commentId).author.email.equals(#auth.name) or hasRole('ADMIN')")
```
создатель комментария, а не объявления

Reply:
и поле в сущности юзера называется теперь username, а не email